release.py

```diff
@@ -0,0 +1,848 @@
+# Copyright 2025 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Release automation script."""
+
+from __future__ import annotations
+
+import argparse
+import logging
+import os
+import re
+import subprocess
+import time
+import webbrowser
+from datetime import datetime
+from pathlib import Path
+
+import github
+import github.GitRelease
+import github.Repository
+
+
+class SymbolFormatter(logging.Formatter):
+    """Custom logging formatter to use symbols for log levels."""
+
+    from typing import ClassVar
+
+    levels: ClassVar[dict[str, str]] = {
+        'DEBUG': '🐛',
+        'INFO': 'ℹ️',  # noqa: RUF001
+        'WARNING': '⚠️',
+        'ERROR': '❌',
+        'CRITICAL': '🔥',
+    }
+
+    def format(self, record: logging.LogRecord) -> str:
+        """Format the log record with symbols for level."""
+        level = self.levels.get(record.levelname, '#')
+        message = record.getMessage()
+        return f'{level} {message}'
+
+
+logger = logging.getLogger('release')
+logger.setLevel(logging.DEBUG)
+handler = logging.StreamHandler()
+handler.setFormatter(SymbolFormatter())
+logger.addHandler(handler)
+
+
+if 'GITHUB_TOKEN' not in os.environ:
+    logger.critical('Environment variable GITHUB_TOKEN not set.')
+    exit(1)
+
+auth = github.Auth.Token(os.environ['GITHUB_TOKEN'])
+gh_client = github.Github(auth=auth)
+
+
+VERSION_REGEX = r'(\d+\.\d+\.\d+(?:\.dev\d+)?)'
+VERSION_FILES = {
+    'ops/src': 'ops/version.py',
+    'ops/pyproject': 'pyproject.toml',
+    'testing': 'testing/pyproject.toml',
+    'tracing': 'tracing/pyproject.toml',
+    'uvlock': 'uv.lock',
```

Comment:
Speaking of `pathlib`, maybe it would pay to make these `pathlib.Path` where they're defined?
```suggestion
VERSION_FILES = {
    'ops/src': Path('ops/version.py'),
    'ops/pyproject': Path('pyproject.toml'),
    'testing': Path('testing/pyproject.toml'),
    'tracing': Path('tracing/pyproject.toml'),
    'uvlock': Path('uv.lock'),
```